### PR TITLE
fix(Revert): sales order status for order type 'Maintenance

### PR DESCRIPTION
**Pull request by Anurag810**
Reverts #17119
The PR is needed to Revert because Order type Maintenance need delivery note in some cases and is not needed in some cases. So we have decided to create a checkbox Delivery Note required in Sales Order if Order Type is Maintenance

### CODE REVIEW

**Comment by rohitwaghchaure** [comment]
@Anurag810 at JS side it's still hard coded
https://github.com/frappe/erpnext/blob/hotfix/erpnext/selling/doctype/sales_order/sales_order_list.js#L34

**Comment by Anurag810** [comment]
> @Anurag810 at JS side it's still hardcoded
> https://github.com/frappe/erpnext/blob/hotfix/erpnext/selling/doctype/sales_order/sales_order_list.js#L34
It is a standard option, if i have to remove it so how can I distinguish between other order types.

File: erpnext/selling/doctype/sales_order/sales_order.js [hunk 1]
```diff
@@ -181,6 +181,8 @@ erpnext.selling.SalesOrderController = erpnext.selling.SellingController.extend(
 				// maintenance
 				if(flt(doc.per_delivered, 2) < 100 &&
 						["Sales", "Shopping Cart"].indexOf(doc.order_type)===-1) {
+					this.frm.add_custom_button(__('Delivery'),
+						function() { me.make_delivery_note_based_on_delivery_date(); }, __("Make"));
```
Comment (rohitwaghchaure) [inline erpnext/selling/doctype/sales_order/sales_order.js]: Why double? this code is already there at line no 145

Comment (Anurag810) [reply erpnext/selling/doctype/sales_order/sales_order.js]: done

**Review by rohitwaghchaure — APPROVED** [review]

**Pull request merged**